pymeasure/instruments/keithley/keithleyDAQ6510.py

```diff
@@ -0,0 +1,286 @@
+        but increases the number of usable digits. """
+    )
+
+    ####################
+    # Methods        #
+    ####################
+
+    def measure_resistance(self, nplc=1, resistance=100e6, auto_range=True):
+        """ Configures the measurement of resistance.
+
+        :param nplc: Number of power line cycles (NPLC) from 5E-4 to 15 (60 Hz)
+                     or 12 (50 Hz or 400 Hz).
+        :param resistance: Upper limit of resistance in Ohms, from 10 to 100E6 (2-wire),
+                           1 to 100E6 (4-wire with OCOM off), or 1 to 10E3 (4-wire with OCOM on).
+        :param auto_range: A boolean value to enable auto_range if ``True``,
+                           else uses the set resistance.
+        """
+        log.info(f"{self.name} is measuring resistance.")
+        self.write(f":SENS:FUNC \"RES\";:SENS:RES:NPLC {nplc};")
+        if auto_range:
+            self.write(":SENS:RES:RANG:AUTO ON;")
+        else:
+            self.resistance_range = resistance
+        self.check_errors()
+
+    def measure_voltage(self, nplc=1, voltage=1000, auto_range=True):
+        """ Configures the measurement of voltage.
+
+        :param nplc: Number of power line cycles (NPLC) from 5E-4 to 15 (60 Hz)
+                     or 12 (50 Hz or 400 Hz).
+        :param voltage: Upper limit of voltage in Volts, from 100E-3 to 1000 V (DC)
+                        or 100E-3 to 750 V (AC).
+        :param auto_range: A boolean value to enable auto_range if ``True``,
+                           else uses the set voltage.
+        """
+        log.info(f"{self.name} is measuring voltage.")
+        self.write(f":SENS:FUNC \"VOLT\";:SENS:VOLT:NPLC {nplc};")
+        if auto_range:
+            self.write(":SENS:VOLT:RANG:AUTO ON;")
+        else:
+            self.voltage_range = voltage
+        self.check_errors()
+
+    def measure_current(self, nplc=1, current=3, auto_range=True):
+        """ Configures the measurement of current.
+
+        :param nplc: Number of power line cycles (NPLC) from 5E-4 to 15 (60 Hz)
+                     or 12 (50 Hz or 400 Hz).
+        :param current: Upper limit of current in Amps, from 10E-6 to 3 A (DC)
+                        or 100E-6 to 3A (AC).
+        :param auto_range: A boolean value to enable auto_range if ``True``,
+                           else uses the set current.
+        """
+        log.info(f"{self.name} is measuring current.")
+        self.write(f":SENS:FUNC \"CURR\";:SENS:CURR:NPLC {nplc};")
+        if auto_range:
+            self.write(":SENS:CURR:RANG:AUTO ON;")
+        else:
+            self.current_range = current
+        self.check_errors()
+
+    def no_errors(self):
+        """
+        Check to see if the instrument has any errors returned or not.
+
+        :return: ``True`` if there are no errors, ``False`` if there are errors.
+        """
+        return len(self.check_errors() == 0)
```

Comment:
Did you know, that `bool([]) is False`? That means, you can use `if inst.check_errors(): handle_errors()`?

That would make that method superfluous.

A general suggestion: try to word a boolean property, that it works like a question: `no_errors() is True` (or `something_enabled = True`, as that makes it easier to interpret True/False.
In such a case (like your name), you can write in the docstring: `:return: whether there are errors.`. (or maybe `:return bool: whether there are errors.`

Reply:
Thanks—I just removed the method.